### PR TITLE
feature: add homepage

### PR DESCRIPTION
**Pull request by cdrage**
feature: add homepage

### What does this PR do?

Adds the following:
* A homepage that lists all the current builds
* Deleting one build
* Deleting multiple builds / bulk deletion
* Be able to click and open the build output directory

### Screenshot / video of UI

https://github.com/containers/podman-desktop-extension-bootc/assets/6422176/3151f45d-2e5a-4eaa-b384-22d7c353705c

### What issues does this PR fix or reference?

Closes https://github.com/containers/podman-desktop-extension-bootc/issues/196

### How to test this PR?

1. Start a build
2. View it in the dashboard
3. Delete the build
4. Try building another build..
5. Click on the folder button (should open in default file manager on
   your system)

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

### CODE REVIEW

**Comment by cdrage** [comment]
Requires https://github.com/containers/podman-desktop-extension-bootc/pull/192 to be merged (based upon that PR)

**Comment by cdrage** [comment]
Ready for review 👍 @deboer-tim

File: packages/frontend/src/lib/BootcFolderColumn.svelte [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import { BootcBuildInfo } from '/@shared/src/models/bootc';
+
+export let object: BootcBuildInfo;
+
+function openFolder(folder: string) {
+  bootcClient.openFolder(folder);
+}
+</script>
+
+<button class="hover:cursor-pointer flex flex-col" on:click="{() => openFolder(object.folder)}">
```
Comment (deboer-tim) [inline packages/frontend/src/lib/BootcFolderColumn.svelte:12]: Should really use a Link. Either way, violet is one of the 'historic' colors that we aren't supposed to use.

File: packages/frontend/src/lib/BootcStatusIcon.svelte [hunk 2]
```diff
@@ -0,0 +1,44 @@
+<script lang="ts">
```
Comment (deboer-tim) [inline packages/frontend/src/lib/BootcStatusIcon.svelte:1]: Why have copies of upstream for star, spinner, but then not StatusIcon? Seems like we should just use it here.

File: packages/shared/src/models/bootc.ts [hunk 3]
```diff
@@ -29,3 +29,5 @@ export interface BootcBuildInfo {
 }
 
 export type BootcBuildStatus = 'running' | 'creating' | 'success' | 'error' | 'lost' | 'deleting';
+
+export const BOOTC_BUILD_LOG_FILENAME = 'image-build.log';
```
Comment (deboer-tim) [inline packages/shared/src/models/bootc.ts:33]: Leftover from previous commit? I thought we removed this and is not being used by frontend.

File: packages/frontend/src/List.svelte [hunk 4]
```diff
@@ -0,0 +1,152 @@
+<script lang="ts">
```
Comment (deboer-tim) [inline packages/frontend/src/List.svelte:1]: Maybe a more descriptive name like `Homepage.svelte`?

File: packages/frontend/src/lib/BootcEmptyScreen.svelte [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<script lang="ts">
+import EmptyScreen from './upstream/EmptyScreen.svelte';
+import BootcIcon from './BootcIcon.svelte';
+</script>
+
+<EmptyScreen
+  icon="{BootcIcon}"
+  title="No past bootc build available"
```
Comment (deboer-tim) [inline packages/frontend/src/lib/BootcEmptyScreen.svelte:8]: To be more consistent with PD just 'No bootc builds' or 'No bootable container builds'?

File: packages/frontend/src/List.svelte [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+  folderColumn,
+  new Column<BootcBuildInfo>('Actions', { align: 'right', renderer: BootcColumnActions, overflow: true }),
+];
+
+const row = new Row<BootcBuildInfo>({
+  selectable: _build => true,
+});
+</script>
+
+<NavPage bind:searchTerm="{searchTerm}" title="BootC (Bootable Container) Image Builder" searchEnabled="{true}">
```
Comment (deboer-tim) [inline packages/frontend/src/List.svelte:118]: The title is long, and maybe even a bit limiting to just the image builder. Maybe just 'Bootable Containers' like the extension?

File: packages/backend/src/api-impl.ts [hunk 7]
```diff
@@ -34,6 +38,54 @@ export class BootcApiImpl implements BootcApi {
     return buildDiskImage(build, this.history);
   }
 
+  async deleteBuild(build: BootcBuildInfo): Promise<void> {
```
Comment (deboer-tim) [inline packages/backend/src/api-impl.ts:41]: IMHO it feels like overkill to have and expose two separate functions for deleting one vs many. I'd prefer to just keep one as we have in PD. e.g. easy enough to do `deleteBuilds([one])` when necessary and have a single delete implementation.

File: packages/frontend/src/lib/BootcCurrentProgress.svelte [hunk 8]
```diff
@@ -0,0 +1,68 @@
+<script lang="ts">
+import type { BootcBuildInfo, BootcBuildStatus } from '/@shared/src/models/bootc';
+import { bootcClient } from '../api/client';
+
```
Comment (deboer-tim) [inline packages/frontend/src/lib/BootcCurrentProgress.svelte:4]: TBH I really don't like the progress column, feels like duplication with status and odd to have a badge here. I am ok merging for now though, I think once we have something like build names we could refactor or look at other options.

**Review by deboer-tim — CHANGES_REQUESTED** [review]

File: packages/frontend/src/lib/BootcStatusIcon.svelte (hunk 2 above)
Comment (cdrage) [reply packages/frontend/src/lib/BootcStatusIcon.svelte:1]: Reason is that the status that we use for history is different than the pods / containers / kubernetes concepts.

We need to have other statuses such as "error" or "lost" or "success" compared to StatusIcon which has:

```
$: solid = status === 'RUNNING' || status === 'STARTING' || status === 'USED' || status === 'DEGRADED';
```

**Comment by cdrage** [comment]
Ready for another review!


https://github.com/containers/podman-desktop-extension-bootc/assets/6422176/ff791b8a-b464-4c6b-9013-066698f39269



Some notes:
- Suggested changes implemented
- Progress column now removed
- Scope reduced (no more leftover api calls such as "navigate to container")
- Keeping it nice and super simple

**Pull request merged**